### ISSUE TITLE
Possible issue with existing encrypted databases with MASTER source

### ISSUE THREAD

**Issue by grumpy235** [moneymanagerex/moneymanagerex#6770]
MMEX version:
 - [ ] 1.7.0
 - [ ] 1.8.0
 - [ ] 1.8.1
 - [ ] 1.8.2
 - [X ] Other (Current Master)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

Just a heads-up after I compiled the latest MASTER as follows

-- MMEX configuration summary
-- ==========================
-- Version        : 1.8.1-Beta.1
-- Commit         : 83ad0331f (2024-05-21)
-- Branch         : master
-- Host system    : Windows AMD64
-- Target system  : Windows AMD64
-- Build types    : Debug;Release;MinSizeRel;RelWithDebInfo
-- Generator      : Visual Studio 17 2022
-- Toolset        : v143
-- Toolset host   : x64
-- Platform       : x64
-- VC install dir : C:\Program Files\Microsoft Visual Studio\2022\Community\VC\
-- Run-time DLLs  : msvcp140 msvcp140_1 msvcp140_2 msvcp140_atomic_wait msvcp140_codecvt_ids vcruntime140_1 vcruntime140 concrt140 api-ms-win-core-console-l1-1-0 api-ms-win-core-console-l1-2-0 api-ms-win-core-datetime-l1-1-0 api-ms-win-core-debug-l1-1-0 api-ms-win-core-errorhandling-l1-1-0 api-ms-win-core-fibers-l1-1-0 api-ms-win-core-file-l1-1-0 api-ms-win-core-file-l1-2-0 api-ms-win-core-file-l2-1-0 api-ms-win-core-handle-l1-1-0 api-ms-win-core-heap-l1-1-0 api-ms-win-core-interlocked-l1-1-0 api-ms-win-core-libraryloader-l1-1-0 api-ms-win-core-localization-l1-2-0 api-ms-win-core-memory-l1-1-0 api-ms-win-core-namedpipe-l1-1-0 api-ms-win-core-processenvironment-l1-1-0 api-ms-win-core-processthreads-l1-1-0 api-ms-win-core-processthreads-l1-1-1 api-ms-win-core-profile-l1-1-0 api-ms-win-core-rtlsupport-l1-1-0 api-ms-win-core-string-l1-1-0 api-ms-win-core-synch-l1-1-0 api-ms-win-core-synch-l1-2-0 api-ms-win-core-sysinfo-l1-1-0 api-ms-win-core-timezone-l1-1-0 api-ms-win-core-util-l1-1-0 api-ms-win-crt-conio-l1-1-0 api-ms-win-crt-convert-l1-1-0 api-ms-win-crt-environment-l1-1-0 api-ms-win-crt-filesystem-l1-1-0 api-ms-win-crt-heap-l1-1-0 api-ms-win-crt-locale-l1-1-0 api-ms-win-crt-math-l1-1-0 api-ms-win-crt-multibyte-l1-1-0 api-ms-win-crt-private-l1-1-0 api-ms-win-crt-process-l1-1-0 api-ms-win-crt-runtime-l1-1-0 api-ms-win-crt-stdio-l1-1-0 api-ms-win-crt-string-l1-1-0 api-ms-win-crt-time-l1-1-0 api-ms-win-crt-utility-l1-1-0 ucrtbase
-- Install prefix : D:/Projects/mmex_master/build/install
-- Portable state : OFF
-- DB encryption  : ON

-- Versions
-- --========--
-- Windows 10.0.22631
-- CMake 3.29.3
-- MSBuild 17.10.4.21802
-- Visual Studio 17.0
-- MSVC 19.40.33808.0
-- Win SDK 10.0.22621.0
-- wxWidgets 3.2.4
-- wxSQLite3 4.9.1
-- Lua 5.3.5
-- curl 8.8.0
-- gettext 0.21
-- ------------------ 8< -----------------
-- Configuring done (40.5s)
-- Generating done (0.6s)
-- Build files have been written to: D:/Projects/mmex_master/build

**Using my MASTER version of MMEX**
1. Open MMEX and create test db.
2. Convert to encrypted with simple password (qwerty) so no chance of typo error.
3. Close application and re-open the test encrypted db and it works fine.
4. Close application and re-open my live encrypted database and get password prompt.
5. Cut and paste password for this database to ensure no typos - get password error. 

**Using MMEX 1.8.0**
1. My live encrypted database still works fine
2. The test encrypted database now gives password error 

I doubt that this is due to my compiling with VS 17.0 and I am guessing that this probably due to the default cipher changes commited just after 1.8.0 went live.
If so then the next live release will need the user to decrypt with the old release of MMEX and a re-encrypt with the new release as part of the upgrade - this worked for me

**Comment by guanlisheng** [comment]
hi @grumpy235 
this is true as we just switched the default cipher to AES256 (via SQLCipher) from AES128 per https://github.com/moneymanagerex/moneymanagerex/pull/6532

Besides decryption, we can also leverage [wxSQLitePlus](https://github.com/guanlisheng/wxsqliteplus) to re-encrypt to AES256 directly. refer to: https://forum.moneymanagerex.org/viewtopic.php?p=24957#p24957

**Comment by grumpy235** [comment]
@guanlisheng
The joys of modern day computing, trying to be all things to all men or in this case Android, think I prefered the days of punched cards or paper tape and a single proprietary operating system 😆 
Perhaps this issue can be used as a reminder that some thought needs to be taken when delivering the next release. The scripted process for forcing a DB version change still causes the odd problem for non-computer literate users so a scripted solution for cipher change would be ideal.

**Comment by guanlisheng** [comment]
Agreed. I also noticed the wide impact due to db version change, while encryption/cipher would be smaller here as audience scale and not complex/complicated (the file as a whole).
